src/averageAges.js

```diff
@@ -15,17 +15,33 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  let counter = 0;
+  let diedInThisCentury;
+  let currentCentury = century;
```

Comment:
```suggestion
```

Let's get rid of all these variables and simplify your code. You can filter (as you did with children) all men according to the `century` argument and then count age difference and reduce that array.

Like:

```
const men = people.filter(person =>
    century
      ? Math.ceil(person.died / 100) === century && person.sex === 'm'
      : (person.sex === 'm')
    )
```